sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -147,6 +166,16 @@ case class Origin(
       builder.result()
     }
   }
+
+  /** Gets the textual fragment of a SQL query. */
+  def getFragment: String = {
+    if (sqlText.isEmpty || startIndex.isEmpty || stopIndex.isEmpty ||
+      startIndex.get < 0 || stopIndex.get >= sqlText.get.length || startIndex.get > stopIndex.get) {
+      ""
+    } else {
+      sqlText.get.substring(startIndex.get, stopIndex.get)
```

Comment:
We should do truncation to avoid big strings.

Reply:
Not sure about this. We have truncated the output in the **summary** already when show it to users. But the method returns the SQL fragment as part of machine readable form. And we can confuse peers if we returns a fragment of a fragment. And there are no benefits of memory usage - `sqlText` has been allocated already.